### PR TITLE
Resolves strategies from `Annotated` type

### PR DESCRIPTION
**Pull request by sobolevn**
Refs #2978

### CODE REVIEW

**Comment by sobolevn** [comment]
I want to see how this behaves accross multiple Python versions. SInce it works for 3.8

**Comment by sobolevn** [comment]
Ok, @Zac-HD there's a small problem: https://github.com/HypothesisWorks/hypothesis/pull/3082/checks?check_run_id=3507228624

It fails because this line here https://github.com/HypothesisWorks/hypothesis/blob/c4fb7c28ba7af68c7f78cb01c4642cdb51d826cc/hypothesis-python/src/hypothesis/strategies/_internal/core.py#L877 uses `get_type_hints()` directly, which returns `{x: integers()}` in 3.9

**Comment by Zac-HD** [comment]
> uses `get_type_hints()` directly, which returns `{x: integers()}` in 3.9

Nope, that's not the reason!  It's failing because without passing `get_type_hints` will only return the type-being-annotated unless you use the new `include_extras=True` argument, which I've now added 😁

File: hypothesis-python/src/hypothesis/internal/compat.py [hunk 1]
```diff
@@ -81,28 +81,25 @@ def get_type_hints(thing):
     Never errors: instead of raising TypeError for uninspectable objects, or
     NameError for unresolvable forward references, just return an empty dict.
     """
+    kwargs = {} if sys.version_info[:2] < (3, 9) else {"include_extras": True}
```
Comment (sobolevn) [inline hypothesis-python/src/hypothesis/internal/compat.py:84]: I was not aware of this hack! Oh wow!

**Review by sobolevn — COMMENTED** [review]
Thanks a lot for your help! I was pretty much stuck with it 🙂

**Review by Zac-HD — APPROVED** [review]
Let's ship it!

**Pull request merged**